src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,42 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService {
+    @Override
+    public List getAll(Class type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        } else if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        } else if (type == Truck.class) {
+            return new TruckProducer().get();
+        } else {
+            return new ArrayList<>();
+        }
+    }
+
+    @Override
+    public void fill(List machines, Machine value) {
```

Comment:
The `fill` method should use `List<? super T>` for the `machines` parameter and `T` for the `value` parameter to match the `MachineService` interface and ensure type safety.